tests/infer/mcmc/test_mcmc.py

```diff
@@ -86,3 +87,22 @@ def test_num_chains(num_chains, cpu_count, monkeypatch):
         assert isinstance(mcmc.sampler, _SingleSampler)
     else:
         assert isinstance(mcmc.sampler, _ParallelSampler)
+
+
+def _empty_model():
+    return torch.ones([])
```

Comment:
One really odd thing that I am noticing is that a `torch.tensor` constructor throws an error under JIT, so if we change this to `torch.tensor(1)` we will get an error when `test_empty_sample_sites` is run under JIT (same for line 150 in hmc.py). This is rather strange, but it is not replicable. It runs fine with the debugger and I am unable to isolate this into a minimal example to present to PyTorch devs.